apps/src/code-studio/pd/form_components/FieldGroup.jsx

```diff
@@ -19,16 +19,31 @@ export default class FieldGroup extends React.Component {
 
   handleChange(event) {
     const value = event.target.value;
+    if (this.props.type === 'number') {
+      // We only want numbers out of this text field, and so this regular expression restricts input to
+      // digits, a single decimal point, and an optional minus sign at the
+      // beginning.
+      if (!value.match(/^$|^-?[0-9]*(\.?)[0-9]*$/)) {
```

Comment:
Why the capture group around the decimal point?

Reply:
Oh, that's just me not knowing any better.  Will fix!